src/basic/fd-util.h

```diff
@@ -126,3 +128,23 @@ static inline char *format_proc_fd_path(char buf[static PROC_FD_PATH_MAX], int f
 
 #define FORMAT_PROC_FD_PATH(fd) \
         format_proc_fd_path((char[PROC_FD_PATH_MAX]) {}, (fd))
+
+/* Determining a file descriptor's path via /proc is not reliable. Only use this function to provide better
+ * log messages when doing fd based path handling! Note that this function uses a thread-local shared buffer
+ * to store the path so the return value will only stay valid until this function is called again from the
+ * same thread. */
+static inline const char *FORMAT_FD_PATH(int fd) {
+        static thread_local char *buf = NULL;
```

Comment:
how is this memory supposed to be freed?

Reply:
The other thing we could do is just use a limited size buffer on the stack and truncate paths if they're longer than that. e.g. 256 bytes instead of 4k

Reply:
but if you create a lot of threads and leave them you'd each time leak memory. that's no ok!

why not just use dynamic memory + `_cleanup_free_()`?

Reply:
We use `_cleanup_free_` now.